app/controllers/application_controller/explorer.rb

```diff
@@ -10,12 +10,14 @@ def build_replaced_trees(replace_trees, valid_values)
   end
 
   def try_build_tree(tree_symbol)
-    method_name = "build_#{tree_symbol}_tree"
-    return unless respond_to?(method_name, true)
-    build_method = method(method_name)
-    # FIXME: This is temporary, we actually need to remove all the build_*_tree methods and
-    # use the Feature::build_tree instead.
-    build_method.arity == 1 ? build_method.call(tree_symbol) : build_method.call
+    # Legacy support for build_*_tree methods
+    # FIXME: delete this after all of them were converted (remaining: build_reports_tree)
+    method_name = :"build_#{tree_symbol}_tree"
+    if respond_to?(method_name, true)
+      method(method_name).call
+    else
+      features.find { |f| f.name == tree_symbol }.try(:build_tree, @sb)
```

Comment:
could you explain this addition?

2nd commit (modifying this function) made sense but this was added - just curious why

Reply:
Yup, it finds the right item in the `featues` hash and tries to call its `build_tree` method, see `Feature#build_tree`.